CorpusCallosum/fastsurfer_cc.py

```diff
@@ -0,0 +1,880 @@
+        Trained model for AC-PC detection.
+    slices_to_analyze : int
+        Number of slices to process.
+
+    Returns
+    -------
+    ac_coords : np.ndarray
+        Coordinates of the anterior commissure.
+    pc_coords : np.ndarray
+        Coordinates of the posterior commissure.
+    """
+
+    # get center of third ventricle from aseg and map to fsaverage space
+    third_ventricle_mask = aseg_nib.get_fdata() == 4
+    third_ventricle_center = np.argwhere(third_ventricle_mask).mean(axis=0)
+    third_ventricle_center_vox = apply_transform_to_pt(third_ventricle_center, orig_fsaverage_vox2vox, inv=False)
+
+    # get 5 mm of slices output with 3 slices per inference
+    midslices_middle = midslices.shape[0] // 2
+    middle_slices_localization = midslices[
+        midslices_middle - slices_to_analyze // 2 - 1 : midslices_middle + slices_to_analyze // 2 + 2
+    ]
+    ac_coords, pc_coords = localization_inference.run_inference_on_slice(
+        model_localization, middle_slices_localization, third_ventricle_center_vox[1:]
+    )
+
+    return ac_coords, pc_coords
+
+
+def segment_cc(
+    midslices: np.ndarray,
+    ac_coords: np.ndarray,
+    pc_coords: np.ndarray,
+    aseg_nib: "nib.Nifti1Image",
+    model_segmentation: "torch.nn.Module",
+    slices_to_analyze: int
+) -> tuple[np.ndarray, np.ndarray]:
+    """Segment the corpus callosum using a trained model.
+
+    Performs corpus callosum segmentation on mid-sagittal slices using a trained model,
+    with AC-PC points as anatomical references. Includes post-processing to clean 
+    the segmentation.
+
+    Parameters
+    ----------
+    midslices : np.ndarray
+        Array of mid-sagittal slices.
+    ac_coords : np.ndarray
+        Anterior commissure coordinates.
+    pc_coords : np.ndarray
+        Posterior commissure coordinates.
+    aseg_nib : nibabel.Nifti1Image
+        Subject's segmentation image.
+    model_segmentation : torch.nn.Module
+        Trained model for CC segmentation.
+    slices_to_analyze : int
+        Number of slices to process.
+
+    Returns
+    -------
+    tuple[np.ndarray, np.ndarray]
+        - segmentation : Binary segmentation of the corpus callosum.
+        - outputs_soft : Soft segmentation probabilities.
+    """
+    # get 5 mm of slices output with 9 slices per inference
+    midslices_middle = midslices.shape[0] // 2
+    middle_slices_segmentation = midslices[
+        midslices_middle - slices_to_analyze // 2 - 4 : midslices_middle + slices_to_analyze // 2 + 5
+    ]
+    segmentation, inputs, outputs_avg, outputs_soft = segmentation_inference.run_inference_on_slice(
+        model_segmentation,
+        middle_slices_segmentation,
+        AC_center=ac_coords,
+        PC_center=pc_coords,
+        voxel_size=aseg_nib.header.get_zooms()[0],
+    )
+
+    pre_clean_segmentation = segmentation.copy()
+    segmentation, cc_volume_mask = segmentation_postprocessing.clean_cc_segmentation(segmentation)
```

Comment:
```suggestion
    segmentation, cc_volume_mask = segmentation_postprocessing.clean_cc_segmentation(pre_clean_segmentation)
```

functions should never operate inplace, if that is not VERY obvious, i.e. trailing underscore or inplace keyword argument.